File: app/src/main/java/io/github/prabhuomkar/flare/models/Task.java
package io.github.prabhuomkar.flare.models;

public class Task {
    private String name, description, imageLink;

    public Task(String name, String description, String imageLink) {
        this.name = name;
        this.description = description;
        this.imageLink = imageLink;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public String getImageLink() {
        return imageLink;
    }
}

File: app/src/main/java/io/github/prabhuomkar/torchexpo/data/models/Task.kt
package io.github.prabhuomkar.torchexpo.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverter
import androidx.room.TypeConverters

@Entity(tableName = "tasks")
@TypeConverters(ResearchAreaTypeConverter::class)
data class Task(
    @PrimaryKey
    val id: Int,
    val name: String,
    val description: String,
    val imageLink: String,
    val researchArea: ResearchArea
)

enum class ResearchArea {
    Vision,
    Generative,
    Language,
    Audio,
    Unknown
}

class ResearchAreaTypeConverter {
    @TypeConverter
    fun fromResearchArea(researchArea: ResearchArea): String {
        return researchArea.toString()
    }

    @TypeConverter
    fun toResearchArea(area: String): ResearchArea {
        return when (area) {
            "Vision" -> ResearchArea.Vision
            "Generative" -> ResearchArea.Generative
            "Language" -> ResearchArea.Language
            "Audio" -> ResearchArea.Audio
            else -> ResearchArea.Unknown
        }
    }
}
